pkg/isoeditor/rhcos.go

```diff
@@ -27,12 +32,77 @@ func NewEditor(dataDir string) Editor {
 	return &rhcosEditor{workDir: dataDir}
 }
 
+func CreateNmstateRamDisk() ([]byte, error) {
+	srcPath := "/usr/bin/nmstatectl"
+
+	// Check if the file exists
+	if _, err := os.Stat(srcPath); os.IsNotExist(err) {
+		log.Fatalf("nmstatectl_x86_64_v4.17 source file does not exist: %v", err)
+	}
+
+	nmstateBinContent, err := os.ReadFile(srcPath)
+	if err != nil {
+		fmt.Println("Error reading file:", err)
+		return []byte{}, err
+	}
+
+	// Run gzip compression
+	compressedBuffer := new(bytes.Buffer)
+	gzipWriter := gzip.NewWriter(compressedBuffer)
+
+	// Create CPIO archive
+	cpioWriter := cpio.NewWriter(gzipWriter)
+
+	if err := cpioWriter.WriteHeader(&cpio.Header{
+		Name: srcPath,
+		Mode: 0o100_755,
+		Size: int64(len(nmstateBinContent)),
+	}); err != nil {
+		return nil, errors.Wrap(err, "Failed to write nmstatectl_x86_64_v4.17 binary CPIO header")
+	}
+	if _, err := cpioWriter.Write(nmstateBinContent); err != nil {
+		return nil, errors.Wrap(err, "Failed to write nmstatectl_x86_64_v4.17 binary CPIO archive")
+	}
+
+	if err := cpioWriter.Close(); err != nil {
+		return nil, errors.Wrap(err, "Failed to close nmstatectl_x86_64_v4.17 binary CPIO archive")
+	}
+	if err := gzipWriter.Close(); err != nil {
+		return nil, errors.Wrap(err, "Failed to close gzip writer")
+	}
+
+	padSize := (4 - (compressedBuffer.Len() % 4)) % 4
+	for i := 0; i < padSize; i++ {
+		if err := compressedBuffer.WriteByte(0); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Can we create a helper for this? It's not something that I feel like should be repeated.

Maybe even create some way to dedupe even more of this function and [`Archive`](https://github.com/openshift/assisted-image-service/blob/180a34e65986e17b3ee16861ca40d64ae442f0d6/pkg/isoeditor/ignition.go#L15-L48)?